domain_models/views.py

```diff
@@ -0,0 +1,170 @@
+
+        if not isinstance(exclude, tuple):
+            raise TypeError("Attribute __exclude__ must be a tuple.")
+
+        if all((not include, not exclude)):
+            return None
+
+        if all((include, exclude)):
+            raise AttributeError("Usage of __include__ and __exclude__ "
+                                 "at the same time is prohibited.")
+
+        include_names = [item.name for item in include]
+        exclude_names = [item.name for item in exclude]
+
+        mcs.chk_intersections(include_names, exclude_names)
+        mcs.match_unknown_attrs(include_names, exclude_names)
+
+    @classmethod
+    def chk_intersections(mcs, include, exclude):
+        """Check whether intersections exist."""
+        intersections = mcs.get_intersections(include)
+        attr, intersections = ('__include__', intersections) \
+            if intersections \
+            else ('__exclude__', mcs.get_intersections(exclude))
+
+        if intersections:
+            raise AttributeError(
+                "It is not allowed to mention already defined properties: "
+                "{0} in {1} attributes.".format(", ".join(intersections),
+                                                attr))
+
+    @classmethod
+    def get_intersections(mcs, attr):
+        """Return intersection with defined properties if exists.
+
+        :type attr: list
+        :rtype: list
+        """
+        if not attr:
+            return []
+        return list(set(mcs.get_properties()).intersection(attr))
+
+    @classmethod
+    def match_unknown_attrs(mcs, include, exclude):
+        """Check about using nonexistent attributes."""
+        model_cls = mcs.attributes.get('__model_cls__')
+        unknown_attr = []
+        for item in include:
+            if not hasattr(model_cls, item):
+                unknown_attr.append(item)
+
+        for item in exclude:
+            if not hasattr(model_cls, item):
+                unknown_attr.append(item)
+
+        if unknown_attr:
+            raise AttributeError(
+                "Nonexistent attributes: {0}.".format(
+                    ", ".join(unknown_attr)))
+
+    @classmethod
+    def get_properties(mcs):
+        """Return list of names of defined properties.
+
+        :rtype: list
+        """
+        return [key for key, value in six.iteritems(mcs.attributes) if
+                isinstance(value, property)]
+
+
+@six.add_metaclass(ContextViewMetaClass)
+class ContextView(object):
+    """Contextual view class."""
+
+    __model_cls__ = None
+    __model__ = None
+    __include__ = tuple()
+    __exclude__ = tuple()
+    __fields__ = list()
+
+    def __init__(self, model):
+        """Model validation.
+
+        :type model: DomainModel
+        """
+        if not isinstance(model, self.__model_cls__):
+            raise TypeError("\"{0}\" is not an instance of {1}".format(
+                model, self.__model_cls__))
+
+        self.__model__ = model
+
+        if self.__include__:
```

Comment:
Hm... Is there a chance to move this checks (and all this logic) to meta class? If I don't miss anything, on instance level we can already have a list of attributes that should be fetched from model instance, including all inclusions and exclusions.